core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -0,0 +1,213 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.Editable
+import android.text.Selection
+import android.text.Spannable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.webkit.WebView
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.view.ActionMode
+import org.kiwix.kiwixmobile.core.R
+
+class CompatFindActionModeCallback internal constructor(context: Context) :
+  ActionMode.Callback, TextWatcher, View.OnClickListener {
+  private val tag = "CompatFindActionMode"
+  @JvmField var isActive: Boolean
+
+  @SuppressLint("InflateParams")
+  private val customView: View = LayoutInflater.from(context).inflate(R.layout.webview_search, null)
+  private val editText: EditText
+  private val findResultsTextView: TextView
+  private var webView: WebView? = null
+  private val input: InputMethodManager
+  private var actionMode: ActionMode? = null
+
+  init {
+    editText = customView.findViewById(R.id.edit)
+    editText.setOnClickListener(this)
+    findResultsTextView = customView.findViewById(R.id.find_results)
+    input = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+    isActive = false
+    setText("")
+  }
+
+  fun setActive() {
+    isActive = true
+  }
+
+  fun finish() {
+    actionMode?.finish()
+    webView?.clearMatches()
+  }
+
+  // Place text in the text field so it can be searched for.  Need to press
+  // the find next or find previous button to find all of the matches.
+  fun setText(text: String?) {
+    editText.setText(text)
+    val span: Spannable = editText.text
+    val length = span.length
+
+    // Ideally, we would like to set the selection to the whole field,
+    // but this brings up the Text selection CAB, which dismisses this
+    // one.
+    Selection.setSelection(span, length, length)
+
+    // Necessary each time we set the text, so that this will watch
+    // changes to it.
+    span.setSpan(this, 0, length, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+  }
+
+  // Set the WebView to search.  Must be non null, and set before calling startActionMode.
+  fun setWebView(webView: WebView?) {
+    if (null == webView) {
+      throw AssertionError(
+        "WebView supplied to CompatFindActionModeCallback cannot be null"
+      )
+    }
+    this.webView = webView
+    findResultsTextView.visibility = View.VISIBLE
+    this.webView?.setFindListener { activeMatchOrdinal: Int, numberOfMatches: Int, _: Boolean ->
+      val result: String = when {
+        editText.text.toString().isEmpty() -> {
```

Comment:
the braces are unnecessary on these when clauses and this should be able to fit on 1 line.

The variable `result` could be inlined

Reply:
a02fb0f